lib/merchant_repository.rb

```diff
@@ -89,15 +89,15 @@ def find_invoices(id)
   end
 
   def most_revenue(x)
-    merchants.max_by.first(x) do |merchant|
+    merchants.sort_by do |merchant|
       merchant.revenue
-    end
+    end.reverse.first(x)
   end
 
   def most_items(x)
-    merchants.max_by.first(x) do |merchant|
-      merchant.items.count
-    end
+    merchants.sort_by do |merchant|
```

Comment:
Pass `&:quantity_successful_items` as an argument to `sort_by` instead of a block.